### PR TITLE
[Keyboard] Add Prototypist J01 rev 1 support

### PR DESCRIPTION
**Pull request by Flexerm**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/prototypist/j01/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5450
+#define PRODUCT_ID      0x6A31
+#define DEVICE_VER      0x0001
+#define MANUFACTURER Prototypist
+#define PRODUCT J-01 Rev1
+#define DESCRIPTION PCB for the J-01 Rev1 Keyboard:
```
Comment (drashna) [inline keyboards/prototypist/j01/config.h:27]:
```suggestion
```

File: keyboards/prototypist/j01/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/prototypist/j01/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/prototypist/j01/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/prototypist/j01/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/prototypist/j01/rules.mk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/prototypist/j01/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/prototypist/j01/keymaps/default/keymap.c (hunk 2 above)
Comment (Flexerm) [reply keyboards/prototypist/j01/keymaps/default/keymap.c:17]: GPL2+ Header added - Thanks Drashna

File: keyboards/prototypist/j01/keymaps/via/keymap.c (hunk 3 above)
Comment (Flexerm) [reply keyboards/prototypist/j01/keymaps/via/keymap.c:17]: GPL2+ Header added - Thanks Drashna

**Review by Flexerm — COMMENTED** [review]
Taken on Drashna's suggestions, thank you.

**Review by drashna — APPROVED** [review]

File: keyboards/prototypist/j01/j01.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+// 
+
```
Comment (fauxpark) [inline keyboards/prototypist/j01/j01.h:24]:
```suggestion
```

File: keyboards/prototypist/j01/rules.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/prototypist/j01/rules.mk:25]:
```suggestion
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Flexerm** [comment]
I've accepted and committed the changes set out by Fauxpark.

Thank you.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**